### DISCUSSION TITLE
Managing alpine component outside x-data

### DISCUSSION THREAD

**Discussion by natecorkish** [alpinejs/alpine#2167] [1. Help]
So I am trying to manage the data of an alpine component outside the x-data... for example:

    <div x-data="toastr()" x-show="isOpen">
        <span x-text="title">
        <span x-text="body">
    </div>
    
    <script>
        function toastr() {
            isOpen: false,
            title: '',
            body: '',
            open() {
                this.isOpen = true
            },
            set(data) {
                this.title = data.title
                this.body = data.body
            }
        }
    </script>
    
    
    .... other html
    
    <script>
        toastr().open();
        toastr().set({
            'title': "Hello",
            'body': "Some body",
        });
    </script>

When I call ```toastr()``` for open and set, it doesn't show the toaster and I am assuming it doesn't update the title and body too... How can I achieve this?

**Comment by SimoTod** [comment]
You should use a store.
